### PR TITLE
[ie/Vbox7] Fix extractor

### LINKED ISSUE #1098: [Broken] Vbox7 issues
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running yt-dlp version **2021.09.25**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
[Vbox7] f9840099c3: Downloading JSON metadata
[Vbox7] f9840099c3: Downloading webpage
WARNING: [Vbox7] unable to extract JSON-LD ; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
WARNING: [Vbox7] unable to extract thumbnail URL; please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
[info] f9840099c3: Downloading 1 format(s): 0
[download] Destination: Where there's a wheel there's a way - Japanese unicycle specialist shows off her skills [f9840099c3].mpd
[download] 100% of 58.76KiB in 00:02

```

## Description

What it downloaded from the following link was a useless tiny piece of file: 
https://www.vbox7.com/play:f9840099c3

I also tried some other links from Vbox7 and got the same issue. Please fix! Thank you so very much!

### PR DESCRIPTION
**Pull request by seproDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Workaround for Vbox7 while DASH http ranges are not yet supported. Works by modifying the URL to get direct http and hls urls.

Fixes #1098

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/vbox7.py [hunk 1]
```diff
@@ -53,41 +60,33 @@ class Vbox7IE(InfoExtractor):
...
-            'http://vbox7.com/play:%s' % video_id, video_id, fatal=None)
+        data = self._download_json(
+            'https://www.vbox7.com/aj/player/item/options', video_id,
+            query={'vid': video_id})['options']
 
-        info = {}
+        if '.mp4' in data['src']:
+            base_url = data['src'].rpartition('_')[0]
+        else:
+            base_url = data['src'].rpartition('.')[0]
```
Comment (bashonly) [inline yt_dlp/extractor/vbox7.py:70]: what do some `data['src']` values look like?

Comment (seproDev) [reply yt_dlp/extractor/vbox7.py:70]: This depends on the video. As far as I can tell for older videos:
`"https://media34.vbox7.com/sl/KWDFgB4GVvBbRXryw8afsw/1706738400/24/249bb972c2/249bb972c2_720.mp4"`
For newer videos:
`"https://edge349.vbox7.com/sl/SCfALacEV7f-Cbbs7iqbrQ/1706738400/3e/3e18bac5ad/3e18bac5ad.mpd"`
For removed/broken(?) videos:
`""`

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/vbox7.py [hunk 2]
```diff
@@ -29,19 +30,25 @@ class Vbox7IE(InfoExtractor):
...
-        'md5': '99f65c0c9ef9b682b97313e052734c3f',
+        'md5': 'da1dd2eb245200cb86e6d09d43232116',
         'info_dict': {
             'id': '249bb972c2',
             'ext': 'mp4',
             'title': 'Смях! Чудо - чист за секунди - Скрита камера',
+            'uploader': 'svideteliat_ot_varshava',
+            'view_count': int,
+            'timestamp': 1360215023,
+            'thumbnail': 'https://i49.vbox7.com/design/iconci/png/noimg6.png',
```
Comment (dirkf) [inline yt_dlp/extractor/vbox7.py:46]: Really? `noimg6.png` isn't a likely name for a valid image. Should it be filtered out?
I get:
```
            'thumbnail': 'https://i49.vbox7.com/o/249/249bb972c20.jpg',
```

Comment (seproDev) [reply yt_dlp/extractor/vbox7.py:46]: I added this to https://github.com/yt-dlp/yt-dlp/pull/8968. I am not sure why the API responded like this when I wrote the extractor. Maybe the image took some time to be generated?
We will have to see how many of the videos remain in ~4 days, when the site will supposedly delete most user-generated content.

**Pull request merged**